plugins/chat/app/services/chat/auto_join_channels.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+module Chat
+  class AutoJoinChannels
```

Comment:
This service is part 1/2 of the main change introduced in this PR.

It efficiently handles automatically joining users.